crates/fuel-core/src/executor.rs

```diff
@@ -881,60 +878,23 @@ where
     }
 
     /// Mark input utxos as spent
-    fn spend_input_utxos<Tx>(
-        &self,
-        tx: &Tx,
-        db: &mut Database,
-        utxo_validation: bool,
-    ) -> ExecutorResult<()>
+    fn spend_input_utxos<Tx>(&self, tx: &Tx, db: &mut Database) -> ExecutorResult<()>
     where
         Tx: ExecutableTransaction,
     {
         for input in tx.inputs() {
             match input {
-                Input::CoinSigned {
-                    utxo_id,
-                    owner,
-                    amount,
-                    asset_id,
-                    maturity,
-                    ..
-                }
-                | Input::CoinPredicate {
-                    utxo_id,
-                    owner,
-                    amount,
-                    asset_id,
-                    maturity,
-                    ..
-                } => {
-                    let block_created = if utxo_validation {
-                        db.storage::<Coins>()
-                            .get(utxo_id)?
-                            .ok_or(ExecutorError::TransactionValidity(
-                                TransactionValidityError::CoinDoesNotExist(*utxo_id),
-                            ))?
-                            .block_created
-                    } else {
-                        // if utxo validation is disabled, just assign this new input to the original block
-                        Default::default()
-                    };
-
-                    db.storage::<Coins>().insert(
-                        utxo_id,
-                        &CompressedCoin {
-                            owner: *owner,
-                            amount: *amount,
-                            asset_id: *asset_id,
-                            maturity: (*maturity).into(),
-                            status: CoinStatus::Spent,
-                            block_created,
-                        },
-                    )?;
+                Input::CoinSigned { utxo_id, .. }
+                | Input::CoinPredicate { utxo_id, .. } => {
+                    // prune utxo from db
+                    db.storage::<Coins>().remove(utxo_id)?;
                 }
                 Input::MessageSigned { message_id, .. }
                 | Input::MessagePredicate { message_id, .. } => {
+                    // mark message id as spent
                     db.storage::<SpentMessages>().insert(message_id, &())?;
```

Comment:
Could you double-check here that the return type is not `Some`? I know that we do this check at the beginning of the execution(that the message is not spent), but it is cheap also to check it here to be sure(maybe someone in the future will remove the check)=)